piezo_web_app/PiezoWebApp/tests/integration_tests/job_status_test.py

```diff
@@ -0,0 +1,65 @@
+import json
+import pytest
+from tornado.httpclient import HTTPClientError
+from tornado.testing import gen_test
+from kubernetes.client.rest import ApiException
+
+from PiezoWebApp.src.handlers.job_status import JobStatusHandler
+from PiezoWebApp.tests.integration_tests.base_integration_test import BaseIntegrationTest
+
+
+# str | The custom resource's group name
+CRD_GROUP = 'sparkoperator.k8s.io'
+
+# str | The custom resource's plural name. For TPRs this would be lowercase plural kind.
+CRD_PLURAL = 'sparkapplications'
+
+# str | The custom resource's version
+CRD_VERSION = 'v1beta1'
+
+
+class TestJobStatusIntegration(BaseIntegrationTest):
+    @property
+    def handler(self):
+        return JobStatusHandler
+
+    @property
+    def standard_request_method(self):
+        return 'GET'
+
+    @gen_test
+    def test_status_is_returned_from_correct_spark_job(self):
+        # Arrange
+        body = {'spark_job': 'test-spark-job', 'namespace': 'default'}
+        kubernetes_response = {'status': {'applicationState': {'state': 'Running'}}}
+        self.mock_k8s_adapter.get_namespaced_custom_object.return_value = kubernetes_response
+        # Act
+        response_body, response_code = yield self.send_request(body)
+        # Assert
+        assert self.mock_k8s_adapter.get_namespaced_custom_object.call_count == 1
+        call_args = self.mock_k8s_adapter.get_namespaced_custom_object.call_args[0]
+        assert call_args[0] == CRD_GROUP
+        assert call_args[1] == CRD_VERSION
+        assert call_args[2] == 'default'
+        assert call_args[3] == CRD_PLURAL
+        assert call_args[4] == 'test-spark-job'
```

Comment:
I think it should be possible to write this simpler...
```
self.mock_k8s_adapter.get_namespaced_custom_object.assert_called_once_with(CRD_GROUP, CRD_VERSION, 'default', CRD_PLURAL, 'test-spark-job')
```

Separating out the call arguments like this has so far only been needed when one of the call arguments is a more complex object, like a dictionary.

Reply:
bbb886e